packages/generator/src/GenerateContext/EnhancedBase.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {
+  ActionTypeApiName,
+  InterfaceTypeApiName,
+  OntologyV2,
+  QueryApiName,
+  SharedPropertyTypeApiName,
+} from "@osdk/gateway/types";
+import * as path from "node:path";
+import type { MinimalFs } from "../MinimalFs.js";
+import type { WireOntologyDefinition } from "../WireOntologyDefinition.js";
+import type { EnhanceCommon } from "./EnhanceCommon.js";
+import { EnhancedAction } from "./EnhancedAction.js";
+import { EnhancedInterfaceType } from "./EnhancedInterfaceType.js";
+import { EnhancedObjectType } from "./EnhancedObjectType.js";
+import { EnhancedQuery } from "./EnhancedQuery.js";
+import { EnhancedSharedPropertyType } from "./EnhancedSharedPropertyType.js";
+
+export abstract class EnhancedBase<T> {
+  protected _common: EnhanceCommon;
+
+  og: T;
```

Comment:
Eventually we should be able to hide the og object but this made it easier to transition.